version: "3.8"
   
services:
    
  web:
    build: .
    command: bash -c "python manage.py migrate  --settings=settings.dev && python manage.py runserver --settings=settings.dev  0.0.0.0:8000"
    volumes:
      - .:/code
    ports:
      - "8000:8000"
    depends_on:
      - db
    restart: on-failure

  db:
    build:
      context: .
      dockerfile: Dockerfile_DB
    volumes:
      - ./pgdata:/var/lib/postgresql/data
      - ./import_foreign_osm_table.sql:/import_foreign_osm_table.sql
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:5432"]
      interval: 30s
      timeout: 10s
      retries: 5
    
  qgis-server:
    image: 3liz/qgis-map-server:3.16
    volumes: 
        - ./provider/qgis/project:/projects
    ports: 
        - 3000:8080
    environment: 
        - QGSRV_SERVER_WORKERS=4
        - QGSRV_LOGGING_LEVEL=WARNING
        - QGSRV_CACHE_ROOTDIR=/projects
        - QGSRV_CACHE_STRICT_CHECK=no
        - QGSRV_CACHE_SIZE=10
        - QGSRV_SERVER_CROSS_ORIGIN=yes
    depends_on:
      - db
      
  elastic:
    image: docker.elastic.co/elasticsearch/elasticsearch:7.10.2
    environment:
      - discovery.type=single-node
      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
    volumes:
      - ./elasticdata:/usr/share/elasticsearch/data
    ports:
      - "9200:9200"
    depends_on:
      - db


    # docker run -e "ELASTICSEARCH_HOSTS=http://host.docker.internal:9200" -p 5601:5601 kibana:7.12.1